examples/example7.py

```diff
@@ -0,0 +1,186 @@
+          (hmccreds_file))
+    sys.exit(1)
+
+hmc = example7["hmc"]
+cpcname = example7["cpcname"]
+partitionname = example7["partitionname"]
+amq_port = example7['amq_port']
+callback = None
+topic = None
+
+cred = hmccreds.get(hmc, None)
+if cred is None:
+    print("Credentials for HMC %s not found in credentials file %s" %
+          (hmc, hmccreds_file))
+    sys.exit(1)
+
+userid = cred['userid']
+password = cred['password']
+
+# Thread-safe handover of notifications between listener and main threads
+NOTI_DATA = None
+NOTI_LOCK = threading.Condition()
+
+class MyListener(object):
+
+    def on_connecting(self, host_and_port):
+        print("Listener: Attempting to connect to message broker")
+        sys.stdout.flush()
+
+    def on_connected(self, headers, message):
+        print("Listener: Connected to broker")
+        sys.stdout.flush()
+
+    def on_disconnected(self):
+        print("Listener: Disconnected from broker")
+        sys.stdout.flush()
+
+    def on_error(self, headers, message):
+        print('Listener: Received an error: %s' % message)
+        sys.stdout.flush()
+
+    def on_message(self, headers, message):
+        global NOTI_DATA, NOTI_LOCK
+        print('Listener: Received a notification')
+        sys.stdout.flush()
+        with NOTI_LOCK:
+            # Wait until main program has processed the previous notification
+            while NOTI_DATA:
+                NOTI_LOCK.wait()
+            # Indicate to main program that there is a new notification
+            NOTI_DATA = headers
+            NOTI_LOCK.notifyAll()
+
+try:
+    print("Using HMC %s with userid %s ..." % (hmc, userid))
+    session = zhmcclient.Session(hmc, userid, password)
+    cl = zhmcclient.Client(session)
+
+    print("Retrieving notification topics ...")
+    topics = session.get_notfication_topics()
+
+    for entry in topics:
+        if entry['topic-type'] == 'job-notification':
+            job_topic = entry['topic-name']
+            break
+
+    conn = stomp.Connection([(session.host, amq_port)], use_ssl="SSL")
+    conn.set_listener('', MyListener())
+    conn.start()
+    conn.connect(userid, password, wait=True)
+
+    sub_id = 42  # subscription ID
+
+    print("Subscribing for job notifications using topic: %s" % job_topic)
+    conn.subscribe(destination="/topic/"+job_topic, id=sub_id, ack='auto')
+
+    print("Finding CPC %s ..." % cpcname)
+    cpc = cl.cpcs.find(name=cpcname)
+    print("Status of CPC %s: %s" % (cpcname, cpc.get_property('status')))
+
+    print("Finding partition %s ..." % partitionname)
+    partition = cpc.partitions.find(name=partitionname)
+    partition_status = partition.get_property('status')
+    print("Status of partition %s: %s" % (partitionname, partition_status))
+
+    if partition_status == 'active':
+        print("Stopping partition %s asynchronously ..." % partitionname)
+        result = partition.stop(wait_for_completion=False)
+    elif partition_status == 'inactive':
```

Comment:
To be more robust this path should not only handle the 'inactive' but also the 'stopped' case:
 elif partition_status in ('inactive', 'stopped'):